File: src/CourseService/Application/Courses/Queries/Dtos/LessonDetailsDto.cs
namespace Courses.Application.Courses.Queries.Dtos;

public record LessonDetailsDto(
    Guid Id,
    string Title,
    string Description,
    int Index,
    TimeSpan? Duration,
    bool IsPreview,
    string? ThumbnailUrl,
    string? VideoUrl
);

File: Frontend/src/features/lessons/types/LessonDetailsDto.ts
/**
 * Backend DTO: Matches LessonDetailsDto from CourseService
 * Used for network layer communication
 */
export interface LessonDetailsDto {
  courseId: string;
  lessonId: string;
  title: string;
  description: string;
  index: number;
  duration: string | null;
  isPreview: boolean;
  thumbnailUrl: string | null;
  videoUrl: string | null;
}
